pyomo/common/config.py

```diff
@@ -9,10 +9,20 @@
 #  ___________________________________________________________________________
 
 import os
+import platform
 import six
 
 from pyutilib.misc.config import ConfigBlock, ConfigList, ConfigValue
 
+if 'PYOMO_CONFIG_DIR' in os.environ:
+    PYOMO_CONFIG_DIR = os.path.abspath(os.environ['PYOMO_CONFIG_DIR'])
+elif platform.system().lower().startswith(('windows','cygwin')):
+    PYOMO_CONFIG_DIR = os.path.abspath(
+        os.path.join(os.environ.get('LOCALAPPDATA', ''), 'Pyomo'))
+else:
+    PYOMO_CONFIG_DIR = os.path.abspath(
+        os.path.join(os.environ.get('HOME', ''), '.pyomo'))
+
```

Comment:
Is there a good reason why the default is ~/.pyomo instead of PYOMO_ROOT_DIR/.pyomo? I feel like putting the default in the pyomo root directory would be better for the many of us who run multiple virtual environments with different versions of Pyomo.

Reply:
Yes. First, `$HOME/.app_name` on *NIX and `%APPDATA%\app_name` on Windows is the standard location for applications to store user-specific data/configuration. Second, PYOMO_ROOT_DIR is not guaranteed to be writable, for example when Pyomo was installis into the system Python (and that site-packages directory should only be writable by root/administrator. 

I see no problem with a follow-on PR that adds calls to `expanduser()` and `expandvars()` to the PYOMO_CONFIG_DIR processing, and injecting PYOMO_ROOT_DIR into the environment. However, there is a circular dependency we will have to work through: currently the CONFIG Dir is set before the module that computes/sets the ROOT dir.